external.sh

```diff
@@ -56,9 +56,45 @@ function run_env() {
     # getting all the effective routes from the NIC
     output=$(az network nic show-effective-route-table --name $name --resource-group $rg --query "value[?nextHopType=='VirtualNetworkGateway' && !contains(addressPrefix, '10.0.0.0/8')]")
     
+    block_10="$space""_10"
+
+    yaml_file="interim-hosting.yaml"
+    externals_json= ./function.sh get '' "api/spaces/$space/blocks/$block_10/externals"
+
+    yq -o=json '.$space.subnets[]' interim-hosting.yaml | jq -c '.' | while read -r subnet; do
+
+        yaml_cidr=$(echo "$subnet" | yq '.cidr')
+        yaml_name=$(echo "$subnet" | yq '.name')
+        yaml_desc=$(echo "$subnet" | yq '.desc')
+
+        # Loop through each external subnet in the API response
+        echo "$externals_json" | jq -c '.[]' | while read -r api_record; do
+            api_cidr=$(echo "$api_record" | jq -r '.cidr')
+            api_name=$(echo "$api_record" | jq -r '.name')
+            
+            # Read the YAML file and process each subnet
+            if [[ "$yaml_cidr" == "$api_cidr" ]]; then # If it exists in the API then update it
```

Comment:
I think we also need another check, at the moment its only checking the exact matching CIDRs, but what about if there is external address which is larger but the new address in yaml is smaller?
IPAM APIs will not allow the smaller one to add